eis_toolkit/conversions/csv_to_geopandas.py

```diff
@@ -0,0 +1,98 @@
+import geopandas
+
+from eis_toolkit.exceptions import (InvalidParameterValueException,
+                                    InvalidWktFormatException,
+                                    InvalidColumnIndexException)
+
+
+def _csv_to_geopandas(  # type: ignore[no-any-unimported]
+    csv: str,
+    indexes: List[int],
+    target_EPSG: int,
+) -> geopandas.GeoDataFrame:
+
+    with open(csv, 'r') as f:
+        has_header = reader.Sniffer().has_header(f.read(1024))
+
+    if has_header:
+        df = pd.read_csv(csv)
+        if len(indexes) == 1:
+            if len(df.columns) < indexes[0]:
+                raise InvalidColumnIndexException
+            column_names = []
+            for row in df:
+                column_names.append(row)
+            geom_column = column_names[indexes[0]]
+            try:
+                geoms = geopandas.GeoSeries.from_wkt(df[geom_column])
+                geodataframe = geopandas.GeoDataFrame(df, crs=target_EPSG, geometry=geoms)
+                return geodataframe
+            except:
+                raise InvalidWktFormatException
+
+        else:
+            if len(df.columns) < indexes[0] or len(df.columns) < indexes[1]:
+                raise InvalidColumnIndexException
+            column_names = []
+            for row in df:
+                column_names.append(row)
+            try:
+                geom_x = column_names[indexes[0]]
+                geom_y = column_names[indexes[1]]
+                geodataframe = geopandas.GeoDataFrame(
+                    df, crs=target_EPSG, geometry=geopandas.points_from_xy(df[geom_x], df[geom_y]))
+                return geodataframe
+            except:
+                raise InvalidParameterValueException
+    else:
+        df = pd.read_csv(csv, header=None)
+        if len(indexes) == 1:
+            if len(df.columns) < indexes[0]:
+                raise InvalidColumnIndexException
+            try:
+                geoms = geopandas.GeoSeries.from_wkt(df[indexes[0]])
+                geodataframe = geopandas.GeoDataFrame(df, crs=target_EPSG, geometry=geoms)
+                return geodataframe
+            except:
+                raise InvalidWktFormatException
+        else:
+            if len(df.columns) < indexes[0] or len(df.columns) < indexes[1]:
+                raise InvalidColumnIndexException
+            try:
+                geodataframe = geopandas.GeoDataFrame(
+                    df, crs=target_EPSG, geometry=geopandas.points_from_xy(df[indexes[0]], df[indexes[1]]))
+                return geodataframe
+            except:
+                raise InvalidParameterValueException
+
+
+def csv_to_geopandas(  # type: ignore[no-any-unimported]
+    csv: str,
+    indexes: List[int],
+    target_EPSG: int,
+) -> geopandas.GeoDataFrame:
+    """
+    Convert CSV file to geopandas DataFrame.
+
+    Usage of single index expects valid WKT geometry.
+    Usage of two indexes expects the X and Y columns of POINT feature(s).
```

Comment:
Should explicitly state that X-coordinate must be the first index and Y-coordinate the second.